examples/hacker-news/app/Stories.re

```diff
@@ -0,0 +1,265 @@
+  hn @bsRecord {
+   jobStories(offset: $offset, limit: $limit) {
+     ... on Story @bsRecord {
+       kids @bsRecord {
+         text
+         by @bsRecord {
+	        id
+         }
+         time
+       }
+       title
+       url
+       score
+       by @bsRecord {
+	      id
+       }
+       time,
+  descendants
+     }
+     ... on Poll @bsRecord {
+       title
+     }
+     ... on Job @bsRecord {
+       title
+     }
+   }
+  }
+}|}
+    ];
+    module New = [%graphql
+      {|query stories($offset: Int!, $limit: Int!) {
+  hn @bsRecord {
+   newStories(offset: $offset, limit: $limit) {
+     ... on Story @bsRecord {
+       kids @bsRecord {
+         text
+         by @bsRecord {
+	        id
+         }
+         time
+       }
+       title
+       url
+       score
+       by @bsRecord {
+	      id
+       }
+       time,
+  descendants
+     }
+     ... on Poll @bsRecord {
+       title
+     }
+     ... on Job @bsRecord {
+       title
+     }
+   }
+  }
+}|}
+    ];
+
+    let toGraphQLClientQuery = (extractStories, query) => {
+      {
+        GraphQLClient.query: query#query,
+        variables: query#variables,
+        parse: json =>
+          Core_kernel.Option.map(query#parse(json)#hn, ~f=extractStories),
+      };
+    };
+
+    let make = (~storyType, ~offset, ~limit, ()) => {
+      switch (storyType) {
+      | Top =>
+        Top.make(~offset, ~limit, ())
+        |> toGraphQLClientQuery(x => x.topStories)
+      | Jobs =>
+        Jobs.make(~offset, ~limit, ())
+        |> toGraphQLClientQuery(x => x.jobStories)
+      | New =>
+        New.make(~offset, ~limit, ())
+        |> toGraphQLClientQuery(x => x.newStories)
+      | Show =>
+        Show.make(~offset, ~limit, ())
+        |> toGraphQLClientQuery(x => x.showStories)
+      };
+    };
+  };
+};
+
+let component = {
+  open Brisk_macos;
+  let component = Brisk.component("Stories");
+  (~children as _: list(unit), ~renderToolbar, ()) =>
+    component(hooks => {
+      let (selectedStoryType, setSelectedStoryType, hooks) =
+        Brisk.Hooks.state(StoryType.Top, hooks);
+      let (selectedStory, setSelectedStory, hooks) =
+        Brisk.Hooks.state(None, hooks);
+      let hooks =
+        Brisk.Hooks.effect(
+          Always,
+          () => {
+            open StoryType;
+            open Toolbar;
+            let storyItem = (~children as _: list(unit), ~storyType, ()) =>
+              SegmentedControl.(
+                <item
+                  isSelected={storyType === selectedStoryType}
+                  onClick={() => setSelectedStoryType(storyType)}>
+                  ...{toString(storyType)}
+                </item>
+              );
+            let segmentedControl = SegmentedControl.segmentedControl;
+            let () =
+              if (selectedStory === None) {
+                renderToolbar(
+                  <toolbar>
+                    <flexibleSpace />
+                    <item>
+                      <segmentedControl>
```

Comment:
naming collisions suck here. Both toolbar and segmentedControl have `items`. I'm not sure how to avoid those.

Reply:
Can we keep module names?

```
<Toolbar.container>
  <Toolbar.item>
    <SegmentedControl.group>
      <SegmentedControl.item>
```

Reply:
We need to have a better PPX for this. We totally should.